### DISCUSSION TITLE
Restoring a backup

### DISCUSSION THREAD

**Discussion by kwridan** [planetscale/discussion#204] [Q&A] [answered]
I've been experimenting with the backup & restore workflow.

- I have a `main` (production) branch
- I initiated a manual backup
- I manually deleted some data in a demo table via running a query ``DELETE from `demo` `` _(simulating an application error that accidentally deleted data)_
- I went to the backup to initiate a restore 
  - It deployed the backup to a branch (e.g. `main-restore`)
  - I connected to it and I can see the data from `demo` table is there from before the deletion 👍 
- I created a deploy request to `main`

The bit I am struggling with, it seems there is no way to merge this deploy request as there are no schema changes.

<img width="892" alt="Screenshot 2022-05-29 at 08 44 05" src="https://user-images.githubusercontent.com/11914919/170857738-30939730-289f-4ca5-813a-4248cd194e6a.png">

I suspect this is similar to https://github.com/planetscale/discussion/discussions/141 - which outlines 2 potential options:

- (1) Manually restoring the data via `pscale database dump` from the restore branch and `pscale database restore-dump` to the main branch
- (2) Swapping branches as described in https://github.com/planetscale/discussion/discussions/141#discussioncomment-1738556

I wasn't able to get (2) working as `main` couldn't be deleted while `main-restore` is a child branch of it. Looking at the follow up comments, it may require making `main-restore` a production branch first. I'm currently experimenting with a hobby plan, so this option won't work as that supports only one production at a time.

I had a few questions as I perhaps misunderstood the purpose of this feature

- Is restoring backups via deploy requests intended for schema changes only?
- For the workflow of swapping branches _(assuming I had the paid plan)_, would that require deploying new connection strings to clients / applications connecting to the database? or is there a way to get a branch agnostic connection string that always points to the default / production branch? _(I recall each connection string is tied to a specific branch)_
- For the manual data dump / restore approach, would that count against the read / write quota?
- Are there any other recommended approaches to restoring a database (including data) from a backup?

Lastly thank you for the great work on PlanetScale - this is mostly me trying to better understand the product and associated workflows 🙂

Thanks!

**Comment by jgreet** [comment, marked as answer]
Hi @kwridan, sorry it took a while to get to your questions. 

>Is restoring backups via deploy requests intended for schema changes only?

Deploy requests are for schema changes only. There isn't a built in way to bring over data from a new (or restored in this case) table in another branch.

>For the workflow of swapping branches (assuming I had the paid plan), would that require deploying new connection strings to clients / applications connecting to the database? or is there a way to get a branch agnostic connection string that always points to the default / production branch? (I recall each connection string is tied to a specific branch)

You're correct that connection strings are branch specific. You would need to create a new connection string for the new branch.

>For the manual data dump / restore approach, would that count against the read / write quota?

Yes, it would count towards both read and write limits. Backups and restores performed within the application do not towards plan limits.

> Are there any other recommended approaches to restoring a database (including data) from a backup?

The simplest flow on the PlanetScale side is to restore the backup to a new branch, promote that branch to production status, and  point your application at it.
